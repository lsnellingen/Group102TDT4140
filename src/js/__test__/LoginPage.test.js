import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from '../pages/LoginPage';
import renderer from 'react-test-renderer';
import { LoginForm, LoginLink } from 'react-stormpath';
import {shallow} from 'enzyme';
import router from 'react-router';

describe('LoginPage', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

    test('renders without crashing', () => {
      /*const renderer = ReactTestUtils.createRenderer();*/
      var tree= shallow(<LoginPage/>);

    /*  const tree = renderer.create(
        <LoginPage/>).toJSON();*/


      expect(tree).toMatchSnapshot();




    });
});
