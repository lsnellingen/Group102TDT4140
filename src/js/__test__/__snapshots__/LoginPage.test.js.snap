// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SideEffect(DocumentTitle)
    title="Login"
>
    <div
        className="container"
    >
        <div
            className="row"
        >
            <div
                className="col-xs-12"
            >
                <h3>
                    Login
                </h3>
                <hr />
            </div>
        </div>
        <LoginForm />
    </div>
</SideEffect(DocumentTitle)>,
  "nodes": Array [
    <SideEffect(DocumentTitle)
      title="Login"
>
      <div
            className="container"
      >
            <div
                  className="row"
            >
                  <div
                        className="col-xs-12"
                  >
                        <h3>
                              Login
                        </h3>
                        <hr />
                  </div>
            </div>
            <LoginForm />
      </div>
</SideEffect(DocumentTitle)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LoginPage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LoginPage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SideEffect(DocumentTitle)
          title="Login"
>
          <div
                    className="container"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-12"
                              >
                                        <h3>
                                                  Login
                                        </h3>
                                        <hr />
                              </div>
                    </div>
                    <LoginForm />
          </div>
</SideEffect(DocumentTitle)>,
        "_debugID": 2,
        "_renderedOutput": <SideEffect(DocumentTitle)
          title="Login"
>
          <div
                    className="container"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-xs-12"
                              >
                                        <h3>
                                                  Login
                                        </h3>
                                        <hr />
                              </div>
                    </div>
                    <LoginForm />
          </div>
</SideEffect(DocumentTitle)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginPage />,
}
`;
